Project_FreeRTOS/main/task_rung.h

```diff
@@ -6,21 +6,63 @@
 
 void task_RUNG(void *pvParameters)
 {
+    const int rung_noiseThreshold = ....; // cập nhật giá trị
+    int rung_unchangedCount=0;
+    int rung_lastValue1 = 0;
+    int rung_lastValue2 = 0;
+    int rung_count=0;
+    unsigned long rung_lastSignalTime = 0;
   for(;;)
   {
+    unsigned long rung_currentTime = millis(); // đọc giá trị thời gian hiện tại
+
+
+    int rung_value = analogRead(IN_SW_1801P);
+    Serial.print("Rung Value: ");
+    Serial.println(rung_value); 
    Message sw_send = {0};
       sw_send.id_Tx = OUT_BUZZER;
       sw_send.id_Rx = IN_SW_1801P;
       sw_send.payload = BUZZER_SW_DISABLE;
       xQueueSend(Platform_Queue, &sw_send, portMAX_DELAY);
-      if(xQueueSend(Platform_Queue, &sw_send, portMAX_DELAY) == pdPASS)
+           if(xQueueSend(Platform_Queue, &sw_send, portMAX_DELAY) == pdPASS)
       {
         // Thực hiện nhiệm vụ của input để đóng gói và gửi đi
+        if(abs(rung_value-rung_lastValue1)>=rung_noiseThreshold)
+        {
+          rung_count++;
+          // ngưỡng báo động
+          if(rung_value > ... && rung_count>=... )// ngưỡng giá trị báo động 
+          {
+            sw_send.payload = BUZZER_SW_ENABLE;
+            Serial.print("Buzzer on");
+            rung_lastValue1=rung_value;
+
+          }
+          else 
+          {
+          sw_send.payload = BUZZER_SW_DISABLE;
+          Serial.print("Buzzer off");
```

Comment:
This one is the same above my comment, we should send the message to queue after updating the payload data